src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,130 @@
+import {
+    Component,
+    Input,
+    ContentChildren,
+    QueryList,
+    Renderer2,
+    ElementRef,
+    Optional,
+    SkipSelf,
+    Output,
+    AfterViewInit,
+    AfterContentInit,
+    OnDestroy
+} from '@angular/core';
+import { Router, NavigationEnd, UrlTree } from '@angular/router';
+import { filter } from 'rxjs/operators/filter';
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+    selector: '[ux-navigation-item]',
```

Comment:
Attribute selectors should follow upper camel case naming convention, eg: `uxNavigationItem`

Reply:
Leave this as is for now, this is a bit of a grey area as components generally don't use an attribute selector. We can change this at a later point if we want to